examples/finnegans-wake-federated.py

```diff
@@ -3,41 +3,56 @@
 
 # WIP w/ hendrix@3.0.0
 
+import binascii
 import datetime
+import logging
 import sys
 
 import maya
-from sandbox_resources import SandboxRestMiddleware
 
 from nucypher.characters import Alice, Bob, Ursula
 from nucypher.data_sources import DataSource
 # This is already running in another process.
 from nucypher.network.middleware import RestMiddleware
 from umbral.keys import UmbralPublicKey
 
-URSULA = Ursula.from_rest_url(network_middleware=RestMiddleware(),
-                              host="localhost",
-                              port=3601,
-                              federated_only=True)
+root = logging.getLogger()
+root.setLevel(logging.DEBUG)
 
-network_middleware = SandboxRestMiddleware([URSULA])
+ch = logging.StreamHandler(sys.stdout)
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+root.addHandler(ch)
+
+teacher_dht_port = 3552
+teacher_rest_port = 3652
+with open("examples-runtime-cruft/node-metadata-{}".format(teacher_rest_port), "r") as f:
+    f.seek(0)
+    teacher_bytes = binascii.unhexlify(f.read())
```

Comment:
You can actually just do `bytes.fromhex(f.read())`, there is no need for using binascii here.